doc/content/devices/adding-devices/_index.md

```diff
@@ -23,103 +23,103 @@ To create a device, first open the application you wish to add the device in. Go
 
-{{< figure src="device-repo.png" alt="Creating a new device with the Device Repository" >}}
+Now just bring your device's QR code closer to your camera. A window with data found from the QR code will appear, containing your device's **Claim authentication code**, **JoinEUI**, **DevEUI** and **Brand**. If the data seems correct, click **Apply**.
+
+{{< figure src="found-qr-data.png" alt="Found QR data" >}}
+
+Now proceed to choose the **Input Method** under the **End device type** section below and keep following the corresponding instructions for [using the LoRaWAN Device Repository]({{< ref "/devices/adding-devices#using-the-lorawan-device-repository" >}}) or [manual registration]({{< ref "/devices/adding-devices#manually-registering-an-end-device" >}}).
+
+The image below shows an example for when a user chooses to **Enter end device specifics manually**. One can notice that parameters like **Frequency plan**, **LoRaWAN version**, **Regional Parameters version** and **AppKey** have to be set manually, and some advanced settings can be adjusted, while the **JoinEUI** and **DevEUI** fields will be pre-filled from the scanned QR code.
+
+{{< figure src="pre-filled-eui-data.png" alt="Registering device with pre-filled EUI data" >}}
+
+### Using the LoRaWAN Device Repository
+
+The [LoRaWAN device repository](https://github.com/TheThingsNetwork/lorawan-devices) contains device profiles, LoRaWAN information, codecs, and more, for many LoRaWAN devices. Using the device repository to add devices in {{% tts %}} automatically uses the correct LoRaWAN version and regional parameters version, which means less information for you to find!
+
+If your device is not in the device repository, see [Manual Registration]({{< ref "/devices/adding-devices#manually-registering-an-end-device" >}}).
+
+To use the device repository, make sure you choose the **Select the end device in the LoRaWAN Device Repository** input method. Then, select the **End device brand**, **Model**, **Hardware Version**, **Software Version**, and **Profile (Region)** for your device.
 
 Choose a **Frequency plan** appropriate for your region. Your device and gateway must use the same frequency plan to communicate.
 
-Enter a **JoinEUI/AppEUI** if provided by your manufacturer.
+{{< figure src="device-repo.png" alt="Creating a new device with the Device Repository" >}}
 
-{{< note >}} Make sure to use a valid DevEUI/JoinEUI. If you device does not have one, you can generate a random one in accordance with the test ranges defined by the [IEEE 802 standards](https://ieee802.org/). {{</ note >}}
+Enter a **JoinEUI/AppEUI** if provided by your manufacturer and click **Confirm**. If it is not provided by the manufacturer and your device is programmable, you can generate a random one in accordance with the test ranges defined by the [IEEE 802 standards](https://ieee802.org/) or use all zeros, just make sure to program the same value into your device. 
```

Comment:
I think we wanted to discourage using all-zero JoinEUIs but I might be wrong, cc @johanstokking